### PR TITLE
Bugfix: Muted users

### PR DESCRIPTION
**Pull request by kev306**
Muted user messages no longer display in room chat
Muted users can no longer interact with players that muted them

### CODE REVIEW

File: assets/scripts/lobby/chat.js [hunk 1]
```diff
@@ -454,11 +454,11 @@ function addToRoomChat(data) {
 }
 
 function isPlayerMuted(username) {
-  if (mutedPlayers.indexOf(username) !== -1) {
-    return true;
+  if (!username) {
+    return false;
   }
```
Comment (vck3000) [inline assets/scripts/lobby/chat.js:459]: Did you need to do this?

File: assets/scripts/lobby/chat.js [hunk 2]
```diff
@@ -454,11 +454,11 @@ function addToRoomChat(data) {
 }
 
 function isPlayerMuted(username) {
-  if (mutedPlayers.indexOf(username) !== -1) {
-    return true;
+  if (!username) {
+    return false;
```
Comment (vck3000) [inline assets/scripts/lobby/chat.js:458]: Did you need to do this?

File: assets/scripts/lobby/chat.js [hunk 3]
```diff
@@ -454,11 +454,11 @@ function addToRoomChat(data) {
...
   }
 
-  return false;
+  return mutedPlayers.indexOf(username.toLowerCase()) !== -1;
```
Comment (vck3000) [inline assets/scripts/lobby/chat.js:461]: Feels like this line alone should be enough

Comment (vck3000) [reply assets/scripts/lobby/chat.js:461]: Can you also double check that when players are muted, their names are lowercased as well?

File: assets/scripts/lobby/chat.js (hunk 2 above)
Comment (kev306) [reply assets/scripts/lobby/chat.js:458]: theres a few sections where the username being passed in is undefined (i.e. when its system messages etc). I can update each one of those areas individually but feel its better to do a sanity check here

File: assets/scripts/lobby/chat.js (hunk 3 above)
Comment (kev306) [reply assets/scripts/lobby/chat.js:461]: The mutedplayers list for players is all in lowercase

**Pull request merged**